src/transformers/modeling_utils.py

```diff
@@ -1464,24 +1464,34 @@ def _get_resized_embeddings(
                 f" {nn.Embedding}."
             )
 
-        # Build new embeddings
-        new_embeddings = nn.Embedding(new_num_tokens, old_embedding_dim)
-        new_embeddings.to(old_embeddings.weight.device, dtype=old_embeddings.weight.dtype)
-
-        # initialize all new embeddings (in particular added tokens)
-        self._init_weights(new_embeddings)
-
-        # Copy token embeddings from the previous weights
-
         # numbers of tokens to copy
         n = min(old_num_tokens, new_num_tokens)
         if is_deepspeed_zero3_enabled():
             import deepspeed
 
-            with deepspeed.zero.GatheredParameters(old_embeddings.weight, modifier_rank=0):
-                if torch.distributed.get_rank() == 0:
```

Comment:
I am not familiar on this topic, but why this line is removed? I am sure there is good reason.

Reply:
I think github diff is a bit misleading here. These two lines were replaced with (see above)
```
            params = [old_embeddings.weight, new_embeddings.weight]
            with deepspeed.zero.GatheredParameters(params, modifier_rank=0):
```
The only thing  I removed was 
```
 if torch.distributed.get_rank() == 0:
```
since it is a parallel process it is not saving latency. I can add it back in if you think it would make more sense.